### DISCUSSION TITLE
Sigma File Manager v1.4

### DISCUSSION THREAD

**Discussion by aleksey-hoffman** [aleksey-hoffman/sigma-file-manager#138] [Announcements]
# Release Notes:
This release improves resource usage, brings a few new features, and a lot of fixes:

- 1 new major feature
- 1 new minor feature
- 18 improvements
- 7 fixes

##  Milestones
🔼 **by v2.0.0 (December 2021) | (Goal: 3 of 20):** Add 20 more new major features

## Download
You can find the download link for your system in the **Asset** section below.

This release contains only Windows build. MacOS and Linux builds will be added in v1.5 update (in a few days).

## New major features
- 574787c: **Feature**: added support for cloud drives (OneDrive, Google Drive, etc.).

## New minor features
- 8401af7: **UX**: new setting: the date format can now be customized. Currently available numeric: `23.05.2021` and short format: `23 May 2021`. It adapts to the local format automatically. For example, in Japan it will be displayed as `2021/5/23` or `2021年5月23日`

## Big improvements
- 92a6fd5:  **UX / optimization**: the background manager for home banner will now display small artwork previews instead of loading original images. This reduced memory usage peaks from `700 MB` to `150 MB`,

## Other improvements
- 0d09def: **UX**: Window transparency effect can now be customized for every page independently.
- ec1e54e: **UX**: Window transparency effect: added settings to control parallax effect.
- ea1c8e9: **UI**: Window transparency effect: added 2 settings to control effect visibility
- 621c886: **UI**: Home banner: added "glow" effect.
- 9b2faf8: **UX**: Upgrade video downloader. It can now extract more video URLs.
- 520789d: **UI**: Info panel: improved performance, added visual improvements, added file status (if offline) and file mime.
- 4b3dd1b: **UX**: Drive cards will now display drive memory type (SSD or HDD).
- ec3f1a4: **UI**: Added new artwork "Slum by Vladimir Manyukhin (animated by Aleksey Hoffman)".
- 7084aae: **UI**: Added new artwork "INCREASE by Sweeper3d (Austin Richey)".
- b9b0664: **UI**: Added new artwork "Cyber Neon City by Laury Guintrand".
- b9b0664: **UI**: Added new artwork "Lip Sync by Han Yang".
- 7802ac8, 5a657e4: **UI**: Slightly improved notifications.
- 590fb02: **UI**: Inaccessible files will now display at least basic info like size and date it was modified and created.
- 05d1786: **UX**: the time of file creation / change will now be displayed along with the date.
- 05d1786: **UX**: the date and time of file creation / change will now be displayed in local format:
**Japan**: 2021年5月23日 23:16
**USA**: May 23, 2021, 11:16 PM
**Russia**: 23 мая 2021 г., 23:16
**Germany**: 23. Mai 2021, 23:16
- b2e210a: **UX**: the app will now show permissions dialog before activating the local share feature for the first time to make sure user understands why the app is asking network permissions.
- ce9a70e: Update dependencies (security fixes)
- A few other small improvements

## Big fixes
- 99f5792: **Fix**: Local network sharing feature is now working properly.
- 4b3dd1b: **Fix**: Drive labels containing non-ASCII characters are now displayed properly.
- 195d683: **Fix**: Window transparency effect now works on all screen sizes.
- 6281a45: **Fix**: the "get size" button in the info panel will now work properly and show the directory size.

## Other fixes
- 28766c1: **Fix**: the error during launch where the app couldn't get the paths to user directories (`~/pictures`, `~/documents`, etc).
- 728a1b2: **Fix**: the page scroll position will now be restored on all pages.
- 1176d7d: **Fix**:  `Linux`: tray icon will no longer duplicate on every page reload and tray menu update.
- A few other small fixes

## Supporters

![Patreon-progress-goal-1](https://user-images.githubusercontent.com/61761672/129417221-95d3c176-54ff-4c53-a285-adcb974c8254.png)

When we meet the goal, I will be able to dedicate more of my time on the app development, and add new features more frequently.

This release was made possible by my Patreon supporters.
Show them some love, check out their work:

<div>
  <a href="https://www.photoancestry.com/" title="Houston Photo Restoration">
    <b>~ Houston Photo Restoration</b>
  </a>
  <a href="https://github.com/andyundso" title="Andy Pfister">
    <b>~ Andy Pfister ~</b>
  </a>
</div>

#### 🎉 Supporter goal: 2 of 30 people joined

Supporters get rewards for supporting this project on Patreon:

- Your issues (feature requests) will have higher priority and a `supporter 💖` label;
- You will be able to vote and decide which features should be prioritized;
- You will get to see Patreon-only posts, and screenshots showcasing new features during development;
- Your name and the link to your project / website will be displayed on this page;
- You will get access to a private Discord where you can learn, get advice, get help, have fun, and just chat with me and other community members;
- Also, thanks to your support, you and everyone else will be getting new features in the app more frequently.

If you like the app, consider supporting my work by joining our community on Patreon. I have a lot of awesome things in development for you guys, but the luck of funding makes it really difficult to create anything.

Check out the full list of rewards on Patreon:

<a target="_blank" href="https://patreon.com/sigma_file_manager">
  <img
    src="https://github.com/aleksey-hoffman/sigma-file-manager/blob/main/.github/media/patreon_button.png"
    width="164px"
  />
</a>

## Installation notes

This is a new app and it doesn't have a signed certificate yet, so you might see a security warning on Windows and MacOS during installation.
- On Windows 10: if you see the Smart Screen prompt, click **"More Info"** and then **"Run Anyway"**. 

<hr /><em>This discussion was created from the release <a href='https://github.com/aleksey-hoffman/sigma-file-manager/releases/tag/v1.4.0'>Sigma File Manager v1.4</a>.</em>

**Comment by Ansh4011** [comment]
### ✅  Update: the issue was fixed

The release asset file was updated after 3b4a4f5 commit
See #139 for more details.

--- 

Hi, I have been trying to install the latest version of SFM and after every install when I launch it, it just gets stuck on the start screen. I have tried updating it via app and also by reinstalling it but none of them worked. I am not really sure what could be the cause though. 
Edit- Facing the same issue in Windows Sandbox as well.

**Reply by aleksey-hoffman** [reply]
@Ansh4011 hi, thanks for the report. Can you please open the app and press `Ctrl + Shift + I` and then click on "console" tab
Do you see any errors there?

**Reply by Ansh4011** [reply]
Yea. This is what it shows-
![image](https://user-images.githubusercontent.com/84930340/132105993-91bb8f23-d758-421d-a55f-1a1d20f2376e.png)

**Reply by aleksey-hoffman** [reply]
@Ansh4011 I changed something and created 2 more builds, can you please try to install both of them and tell me if they work for you?

### Modified version 1:
https://drive.google.com/file/d/1rw4-5rpXolPtAVvvnTOnik7cpFUY-aq7/view?usp=sharing

### Modified version 2:
https://drive.google.com/file/d/1BncjW2Fp-75MrlWLNWOeERWtJIViyiM-/view?usp=sharing

**Reply by Ansh4011** [reply]
Thanks a million for the support :) Will return back to you as soon as possible....

**Reply by aleksey-hoffman** [reply]
Fixed the issue and updated the release asset file after 3b4a4f5 commit
See #139 for more details.

**Comment by davawen** [comment]
No AppImage this release ?

**Reply by aleksey-hoffman** [reply]
@davawen unfortunately yeah, didn't have time to test MacOS and Linux builds.
Struggling with funding, cannot afford to spend more time on the project :(
Hopefully one day enough people will support me on Patreon and I will be able to work on it full time

**Reply by Ansh4011** [reply]
@aleksey-hoffman I am not sure but maybe I'll be getting Linux in a month or two once I get an ssd so whenever I would get free from studies exams and all that stuff (10th is meant to be serious in India lol), I would love to help in the Linux builds :) Being a student I can't help you out with money though, but I would surely be helping out with other stuff. Thanks Bro for SFM, I really personally love it!

**Comment by Makezu90** [comment]


**Reply by aleksey-hoffman** [reply]
I will fix it in the next update in a few days.
I already found out the cause of the issue. It occurs on computers that have only 1 drive (partition) mounted

**Comment by roke-julian-lockhart** [comment]
Is implementation of automation for creation of binary packages for alternative operating-systems difficult?

**Reply by aleksey-hoffman** [reply]
Shouldn't be difficult, you just need to specify the format in `./vue.config.js` and then run `npm run electron:build`.
